packages/paste-core/components/minimizable-dialog/src/MinimizableDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import type {BoxProps} from '@twilio-paste/box';
+import {Box} from '@twilio-paste/box';
+import {styled, css} from '@twilio-paste/styling-library';
+import {StyledBase} from '@twilio-paste/theme';
+import {NonModalDialogPrimitive} from '@twilio-paste/non-modal-dialog-primitive';
+
+import {MinimizableDialogContext} from './MinimizableDialogContext';
+
+const StyledMinimizableDialog = styled.div(({minimized}: {minimized?: boolean}) =>
+  css({
+    boxShadow: 'shadow',
+    width: 'size40',
+    zIndex: 'zIndex80',
+    borderRadius: 'borderRadius20',
+    transform: 'none!important',
+    // Note: not able to use themeGet here, so this needs to change if space70 changes
+    inset: `auto 1.5rem ${minimized === true ? '0' : '1.5'}rem auto!important`,
+    ['&:focus']: {
+      outline: 'none',
+    },
+  })
+);
+
+StyledMinimizableDialog.displayName = 'StyledMinimizableDialog';
+
+export interface MinimizableDialogProps extends Pick<BoxProps, 'element'> {
+  children: React.ReactNode;
+  'aria-label': string;
+}
+
+export const MinimizableDialog = React.forwardRef<HTMLDivElement, MinimizableDialogProps>(
+  ({children, element = 'MINIMIZABLE_DIALOG', ...props}, ref) => {
+    const dialog = React.useContext(MinimizableDialogContext);
+
+    return (
+      <NonModalDialogPrimitive
+        {...(dialog as any)}
```

Comment:
suggestion: I don't think you need this `as any` anymore. The way you extend a partial of non modal dialog primitive props for the context should do the trick. Try it out

Reply:
Nice, that works!